File: js/ControlDigit/Isbn13/Isbn13.js
export function isbn13 (number) {
    let sum = 0;
    let factor = 1;

    do {
        const digit = number % 10;
        sum += factor * digit;
        factor = 4 - factor;
        number = Math.floor(number / 10);
    } while (number > 0);


    let m = sum % 10;
    if (m === 0)
        return 0;

    return 10 - m;
}

File: python/ControlDigit/Isbn13/Isbn13.py
def isbn13(number: int) -> int:
    sum = 0
    factor = 1

    while number > 0:
        digit = number % 10
        sum += factor * digit
        factor = 4 - factor
        number = number // 10

    m = sum % 10
    if m == 0:
        return 0
    return 10 - m
